framework/src/Volo.Abp.EventBus/Volo/Abp/EventBus/UnitOfWorkEventPublisher.cs

```diff
@@ -37,12 +44,44 @@ public async Task PublishDistributedEventsAsync(IEnumerable<UnitOfWorkEventRecor
     {
         foreach (var distributedEvent in distributedEvents)
         {
-            await _distributedEventBus.PublishAsync(
-                distributedEvent.EventType,
-                distributedEvent.EventData,
-                onUnitOfWorkComplete: false,
-                useOutbox: distributedEvent.UseOutbox
-            );
+            if (distributedEvent.UseOutbox)
+            {
+                var eventType = distributedEvent.EventType;
+                var eventData = distributedEvent.EventData;
+
+                if (_abpDistributedEventBusOptions.OutboxExist(distributedEvent.EventType))
+                {
+                    await _distributedEventBus.PublishAsync(
+                        eventType,
+                        eventData,
+                        onUnitOfWorkComplete: false,
+                        useOutbox: true
+                    );
+                }
+                else
+                {
+                    // Fall back to publish after the UOW is completed if the outbox is unavailable.
+                    if (_unitOfWorkManager.Current != null)
+                    {
+                        _unitOfWorkManager.Current.OnCompleted(async() => {
```

Comment:
Now, I understand what you are trying to do. You want to publish event after the UOW completed, transaction is committed. But this is changing the behavior and it is against our design. We intentionally not doing this like that.

(1) With our current design; if we can't publish the event, the transaction is rolled back. If we publish the event, but can't commit the transaction, then we publish an unnecessary event.

(2) If we accept this PR; If we can't publish the event, the transaction is not rolled back. Then we have the operation is done, but we miss the event. Also, the end-user gets an error since we don't hide the exception (if we can't publish the event), but the operation is actually successful. We don't want that.

There is a tradeoff here we choose the option (1). If you don't want the tradeoff, the only solution is to configure the outbox.

So, we want to go with the option (1) and not want to change to option (2). So, sorry, I won't accept this PR.

The only solution we may accept is to provide a framework option to change the behavior for who want it (while keeping the current behavior as default).